coremltools/converters/mil/mil/ops/defs/control_flow.py

```diff
@@ -428,6 +487,29 @@ def value_inference(self):
 
 @register_op(doc_str="")
 class list_write(Operation):
+    """
+    Write a value into index ``index`` of ``ls``.
+    
+    Parameters
+    ----------
+    ls: List (Required)
+    
+    index: <i32> (Required)
+        * Size of the list.
+    
+    value: <*,T> (Optional)
+        * Element value to write, which must match the element shape of ``ls``.
+        * Default is ``None``.
+
+    Returns
+    -------
+    List[*]
+    
+    Attributes
+    ----------
+    T: fp32
```

Comment:
Type domains `i32` & `bool` are also valid for `list` and its operations.

Reply:
Added to list_write, list_read, list_gather, and list_scatter: `T: fp32, i32, bool`